onnxruntime/core/framework/execution_context.cc

```diff
@@ -0,0 +1,335 @@
+#include "core/framework/execution_context.h"
+#include "core/framework/execution_provider.h"
+#include "core/framework/execution_frame.h"
+#include "core/framework/bfc_arena.h"
+#include "core/framework/session_state.h"
+#include "core/common/spin_pause.h"
+
+namespace onnxruntime {
+
+class DeviceStreamCollectionImpl {
+ public:
+  DeviceStreamCollectionImpl(size_t num_streams, const SessionState& sess_state) : num_streams_(num_streams) {
+    device_streams_.resize(num_streams, nullptr);
+    device_streams_containers.reserve(num_streams);
+    auto& providers = sess_state.GetExecutionProviders();
+    eps_.reserve(providers.NumProviders());
+    for (auto& ep : providers) {
+      eps_.push_back(ep);
+    }
+    is_main_graph_ = sess_state.GetGraphViewer().ParentNode() == nullptr;
+  }
+
+  virtual ~DeviceStreamCollectionImpl() {
```

Comment:
Is anything going to derive from this class?

Reply:
no